src/snippet.rs

```diff
@@ -5,56 +5,201 @@
+    pub(crate) id: Option<&'a str>,
+    pub(crate) slices: Vec<Slice<'a>>,
+    pub(crate) footer: Vec<Label<'a>>,
+}
+
+impl<'a> Snippet<'a> {
+    pub fn title(title: Label<'a>) -> Self {
+        Self {
+            title,
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn error(title: &'a str) -> Self {
+        Self {
+            title: Label::error(title),
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn warning(title: &'a str) -> Self {
+        Self {
+            title: Label::warning(title),
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn info(title: &'a str) -> Self {
+        Self {
+            title: Label::info(title),
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn note(title: &'a str) -> Self {
+        Self {
+            title: Label::note(title),
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn help(title: &'a str) -> Self {
+        Self {
+            title: Label::help(title),
+            id: None,
+            slices: vec![],
+            footer: vec![],
+        }
+    }
+
+    pub fn id(mut self, id: &'a str) -> Self {
+        self.id = Some(id);
+        self
+    }
+
+    pub fn slice(mut self, slice: Slice<'a>) -> Self {
+        self.slices.push(slice);
+        self
+    }
+
+    pub fn footer(mut self, footer: Label<'a>) -> Self {
+        self.footer.push(footer);
+        self
+    }
+}
+
+pub struct Label<'a> {
+    pub(crate) annotation_type: AnnotationType,
+    pub(crate) label: &'a str,
+}
+
+impl<'a> Label<'a> {
+    pub fn new(annotation_type: AnnotationType, label: &'a str) -> Self {
+        Self {
+            annotation_type,
+            label,
+        }
+    }
+    pub fn error(label: &'a str) -> Self {
+        Self {
+            annotation_type: AnnotationType::Error,
+            label,
+        }
+    }
+
+    pub fn warning(label: &'a str) -> Self {
+        Self {
+            annotation_type: AnnotationType::Warning,
+            label,
+        }
+    }
+
+    pub fn info(label: &'a str) -> Self {
+        Self {
+            annotation_type: AnnotationType::Info,
+            label,
+        }
+    }
+
+    pub fn note(label: &'a str) -> Self {
+        Self {
+            annotation_type: AnnotationType::Note,
+            label,
+        }
+    }
+
+    pub fn help(label: &'a str) -> Self {
+        Self {
+            annotation_type: AnnotationType::Help,
+            label,
+        }
+    }
+
+    pub fn label(mut self, label: &'a str) -> Self {
+        self.label = label;
+        self
+    }
+
+    /// Create a [SourceAnnotation] with the given span for a [Slice]
+    pub fn span(&self, span: Range<usize>) -> SourceAnnotation<'a> {
```

Comment:
We could generalize this to any range type. up to you

Reply:
I think this is something we could do later if asked for it